File: 25-07/ClientVHShop/DTOs/ProductAddDTO.cs
namespace ChienVHShopOnline.DTOs;

public class ProductAddDTO
{
    public string ProductName { get; set; } = string.Empty;
    public string Image { get; set; } = string.Empty;
     public double? Price { get; set; }
    public int UserId { get; set; }
    public int? CategoryId { get; set; }
    public int? ColorId { get; set; }
    public int? ModelId { get; set; }
    public int? StorageId { get; set; }
    public DateTime? SellStartDate { get; set; }
    public DateTime? SellEndDate { get; set; }
    public int? IsNew { get; set; }
}

File: 25-07/Migration/ShopApp/src/app/models/productadddto.ts
import { ProductModel } from "./product.model";

export class ProductAddDTO {
    constructor(
        public productName: string = "",
        public image: string = "",
        public price: number | null = null,
        public userId: number| null = null,
        public categoryId: number | null = null,
        public colorId: number | null = null,
        public modelId: number | null = null,
        public storageId: number | null = null,
        public sellStartDate: Date | null = null,
        public sellEndDate: Date | null = null,
        public isNew: number | null = null
    ) {}

    static fromModel(product : ProductModel) {
        return new ProductAddDTO(
            product.productName,
            product.image,
            product.price,
            product.userId,
            product.categoryId,
            product.colorId,
            product.modelId,
            product.storageId,
            product.sellStartDate,
            product.sellEndDate,
            product.isNew
        );
    }
}